QtAwesome/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+project(QtAwesome)
+
+add_library(QtAwesome
+  QtAwesome.cpp
+  QtAwesomeAnim.cpp
+  QtAwesomeFree.qrc
+)
+
+target_include_directories(QtAwesome
+  INTERFACE ${PROJECT_SOURCE_DIR}
+)
+
+target_link_libraries(QtAwesome PUBLIC
+  Qt${QT_VERSION_MAJOR}::Widgets
+)
+
+include(GenerateExportHeader)
```

Comment:
I checked this. But it's not a full replace for export_all. It just generates a header with directives. You need to mark exported classes, or you will have something same:
`QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual struct QMetaObject const * __cdecl fa::QtAwesome::metaObject(void)const " (?metaObject@QtAwesome@fa@@UEBAPEBUQMetaObject@@XZ)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual void * __cdecl fa::QtAwesome::qt_metacast(char const *)" (?qt_metacast@QtAwesome@fa@@UEAAPEAXPEBD@Z)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual int __cdecl fa::QtAwesome::qt_metacall(enum QMetaObject::Call,int,void * *)" (?qt_metacall@QtAwesome@fa@@UEAAHW4Call@QMetaObject@@HPEAPEAX@Z)
QtAwesome.obj : error LNK2019: unresolved external symbol "int __cdecl qInitResources_QtAwesomeFree(void)" (?qInitResources_QtAwesomeFree@@YAHXZ) referenced in function "void __cdecl qtawesome_init_resources(void)" (?qtawesome_init_resources@@YAXXZ)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: static struct QMetaObject const fa::QtAwesomeAnimation::staticMetaObject" (?staticMetaObject@QtAwesomeAnimation@fa@@2UQMetaObject@@B)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual struct QMetaObject const * __cdecl fa::QtAwesomeAnimation::metaObject(void)const " (?metaObject@QtAwesomeAnimation@fa@@UEBAPEBUQMetaObject@@XZ)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual void * __cdecl fa::QtAwesomeAnimation::qt_metacast(char const *)" (?qt_metacast@QtAwesomeAnimation@fa@@UEAAPEAXPEBD@Z)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual int __cdecl fa::QtAwesomeAnimation::qt_metacall(enum QMetaObject::Call,int,void * *)" (?qt_metacall@QtAwesomeAnimation@fa@@UEAAHW4Call@QMetaObject@@HPEAPEAX@Z)`

Reply:
Reverted here: 8b3cf90